tribe-ext-select-ticket-providers/index.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Plugin Name:     Event Tickets Extension: Select Ticket Providers
+ * Description:     Adds some checkboxes to WP Admin > Events > Settings > Tickets for choosing which Ticket Providers you want to use.
+ * Version:         1.0.0
+ * Extension Class: Tribe__Extension__Select_Ticket_Providers
+ * Author:          Modern Tribe, Inc.
+ * Author URI:      http://m.tri.be/1971
+ * License:         GPLv2 or later
+ * License URI:     https://www.gnu.org/licenses/gpl-2.0.html
+ */
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) {
+	return;
+}
+
+/**
+ * Extension main class, class begins loading on init() function.
+ */
+class Tribe__Extension__Select_Ticket_Providers extends Tribe__Extension {
+
+	/**
+	 * Contains the full list of providers from before we removed any
+	 *
+	 * @var array
+	 */
+	protected $default_providers;
+
+	/**
+	 * Option key for which providers are enabled
+	 *
+	 * @var string
+	 */
+	protected $option_key_enabled_providers = 'tickets_enabled_providers';
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->add_required_plugin( 'Tribe__Tickets__Main' );
+		$this->add_required_plugin( 'Tribe__Tickets_Plus__Main' );
+
+		$this->set_url( 'https://theeventscalendar.com/extensions/select-ticket-providers' );
+	}
+
+	/**
+	 * Extension initialization and hooks.
+	 */
+	public function init() {
+		add_action( 'init', array( $this, 'remove_providers' ), 100 );
+		add_action( 'admin_init', array( $this, 'add_settings' ) );
+	}
+
+	/**
+	 * Add settings to tribe settings page
+	 */
+	public function add_settings() {
+		require_once dirname( __FILE__ ) . '/src/Tribe/Settings_Helper.php';
+
+		// Setup fields on the settings page
+		$setting_helper = new Tribe__Settings_Helper();
```

Comment:
This should probably be `new Tribe__Extension__Settings_Helper()` to match the name of the class defined in `src/Tribe/Settings_Helper.php` (or else a change is needed there).